### DISCUSSION TITLE
Audience not returning in token when trying to set up OIDC with Ory Network

### DISCUSSION THREAD

**Discussion by aman-hh** [ory/hydra#3562] [Q&A]
Hey there,
I'm trying to set up Ory Network with a server lifecycle management tool built with RoR called [Foreman](https://theforeman.org). I follow the steps outlined [here](https://theforeman.org/manuals/3.6/index.html#5.7.7SingleSign-onforForemanusingOpenIDConnectprotocol) to set up OIDC. Everything seems to be working more or less but the `aud` field is coming back as an empty array even though I'm adding the domain where Foreman is hosted in the OAuth2 configuration settings in the audience field via the Ory Console. Is there something I need to do in order to get the audience claim to return with the token?

**Comment by kmherrmann** [comment]
Did you set the Audience on the OAuth2 Client?
<img width="1379" alt="Screenshot 2023-07-03 at 22 13 33" src="https://github.com/ory/hydra/assets/106238709/ffc0afdf-f822-4c76-9b94-4f81b0b143fe">

... and can you confirm that the audience is returned when getting the client e.g. via CLI:
`ory get oauth2-client $id --project=$project_id `

You then need to make sure the client requests the audience as per https://www.ory.sh/docs/hydra/guides/audiences#audience-in-authorization-code-implicit-and-hybrid-flows

**Reply by aman-hh** [reply]
I set the audience and it's returning via CLI. For the second step, the Foreman application makes requests to whichever URL you set as the issuer, I believe. Should this URL be `/oauth2/auth` with the audience added as a query param, then?

**Reply by kmherrmann** [reply]
yes, I believe so

```
https://{project.slug}.projects.oryapis.com/oauth2/auth
  ?client_id=...
  &scope=...
  &audience=https://api.my-cloud.com/user+https://some-tenant.my-cloud.com/
```

**Reply by aman-hh** [reply]
Using the `/oauth2/auth` endpoint still seems to return an empty audience array. Is there anything else that needs to be set in order to get the audience back in the token?
